### PR TITLE
Update toc using jb toc migrate.

### LINKED ISSUE #64: Unable to build site locally
I've attempted to build the site locally through various methods, including via the `Makefile` and with `jupyter-book`.  So far, I haven't been successful with either approach. When I try to build with the `Makefile` (after manually installing the dependencies to work around #63) I see the following:

```
make[1]: *** book: No such file or directory.  Stop.
make: *** [Makefile:37: html] Error 2
```

I tried working through some of the other steps in the Makefile manually, but haven't been able to progress to getting the site built.

When I try with jupyter-book (e.g. `jb build .`) I see the following:

<details>
  <summary>jb build output</summary>
<pre>
Running Jupyter-Book v0.11.3
Traceback (most recent call last):
  File "/home/ross/.virtualenvs/skimage-tutorials/lib/python3.9/site-packages/jupyter_book/cli/main.py", line 244, in build
    parse_toc_yaml(toc)
  File "/home/ross/.virtualenvs/skimage-tutorials/lib/python3.9/site-packages/sphinx_external_toc/parsing.py", line 82, in parse_toc_yaml
    return parse_toc_data(data)
  File "/home/ross/.virtualenvs/skimage-tutorials/lib/python3.9/site-packages/sphinx_external_toc/parsing.py", line 88, in parse_toc_data
    raise MalformedError(f"toc is not a mapping: {type(data)}")
sphinx_external_toc.parsing.MalformedError: toc is not a mapping: <class 'list'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ross/.virtualenvs/skimage-tutorials/bin/jb", line 8, in <module>
    sys.exit(main())
  File "/home/ross/.virtualenvs/skimage-tutorials/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/ross/.virtualenvs/skimage-tutorials/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/ross/.virtualenvs/skimage-tutorials/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ross/.virtualenvs/skimage-tutorials/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ross/.virtualenvs/skimage-tutorials/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/ross/.virtualenvs/skimage-tutorials/lib/python3.9/site-packages/jupyter_book/cli/main.py", line 246, in build
    _error(
  File "/home/ross/.virtualenvs/skimage-tutorials/lib/python3.9/site-packages/jupyter_book/utils.py", line 48, in _error
    raise kind(box)
RuntimeError: 
===============================================================================

The Table of Contents file is malformed: toc is not a mapping: <class 'list'>
You may need to migrate from the old format, using:

        jupyter-book toc migrate /home/ross/repos/skimage-tutorials/_toc.yml -o /home/ross/repos/skimage-tutorials/_toc.yml

===============================================================================
</pre>
</details>

### PR DESCRIPTION
**Pull request by rossbar**
Closes #64 

At least gets the jupyter-book workflow running. There are still failures in the build process, but it at least gets the build running.

This is the result of running `jb toc migrate` as recommended at the very end of the error message posted in #64 (I missed this in the first readthrough of the error).

### CODE REVIEW

File: _toc.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
-- file: lectures/00_images_are_arrays
+format: jb-article
+root: lectures/00_images_are_arrays
+sections:
```
Comment (jni) [inline _toc.yml:3]:
```suggestion
format: jb-book
root: lectures/00_images_are_arrays
chapters:
```

I think I'd like this repo to more closely resemble a book. See discussion at https://github.com/executablebooks/jupyter-book/issues/1355#issuecomment-857787907

Probably we'll need a different index page but 🤷 that can be a different PR

**Review by jni — APPROVED** [review]
I'm approving, I don't mind whether my comment is incorporated, but I think book is a better default choice. I'll check again tomorrow and merge depending on your response @rossbar!

Thank you!

File: _toc.yml (hunk 1 above)
Comment (rossbar) [reply _toc.yml:3]: I'm not familiar with the distinctions, so I have no opinion! This PR is just the result of running the recommended jupyterbook migrate command, so the output is likely pretty generic.

While we're on the topic - I wanted to gauge interest in investigating a sphinx-based workflow instead. I've had some experience helping to set up related sites for NumPy ([site](https://numpy.org/numpy-tutorials/), [source](https://github.com/numpy/numpy-tutorials/)) and NetworkX ([site](https://networkx.org/nx-guides/), [source](https://github.com/networkx/nx-guides)), and have found the sphinx-based workflow pretty straightforward for setting things up. I know we've had a brief discussion on this elsewhere, but AIUI jupyter-book is a thin wrapper on top of the other lower-level executablebook projects that include sphinx extensions like `myst-nb`. Jupyter-book introduces a layer of indirection for things like the configuration and toc which I believe are designed to be more user-friendly for users who have never used sphinx, but are unnecessary if you are comfortable with a sphinx workflow.

Anyways - I'd be happy to put together a draft PR to illustrate what this could look like, with the understanding that it would only be for illustration purposes and it might not necessarily be adopted. I don't want to step on any toes however: if there is a strong preference for sticking with the current workflow I'm happy to leave off!

Comment (rossbar) [reply _toc.yml:3]: > While we're on the topic - I wanted to gauge interest in investigating a sphinx-based workflow instead.

I completely forgot that @alexdesiqueira has done this already! I'll switch my focus over to #59. Thanks @hmaarrfk for the reminder.

Comment (jni) [reply _toc.yml:3]: > Jupyter-book introduces a layer of indirection for things like the configuration and toc which I believe are designed to be more user-friendly for users who have never used sphinx

🙋 

Honestly I have never grokked sphinx despite "using it" in many projects. I cargo-cult the whole workflow, and I hate it. JB marked the first time that I felt comfortable using a sphinx-based tool.

So, I do actually have a strong preference for using JB *unless* straight sphinx gives a *concrete* advantage that we are lacking. (ie I don't want to hear "configurability", I want to hear "we want to harmonize the styling of the home page and the tutorials and JB makes that very very hard/impossible".)

**Comment by alexdesiqueira** [comment]
Sorry I dropped the ball on #59 everyone. Just answered @jni's review there; please let me know if y'all would like to continue with either this or that.

File: _toc.yml (hunk 1 above)
Comment (stefanv) [reply _toc.yml:3]: I don't quite follow this argument.  If a tool straight out of the box gives you close to what you want, you would usually need justification for an additional dependency.

[Our docs](https://github.com/scikit-image/scikit-image/blob/main/doc/source/conf.py) will probably be the most convoluted Sphinx config you can find, but it is mainly boilerplate Sphinx config.

That all said, we don't even need Sphinx: we just have a bunch of Markdown documents, so Hugo would do the trick just as nicely, and is entirely straightforward to configure.

Comment (jni) [reply _toc.yml:3]: @stefanv "a bunch of markdown documents" — don't we have/want *executable* notebooks (or myst-nb) that need to be *executed* before they are published?

In the future, we probably want hidden exercise solutions, which are [in the works](https://github.com/executablebooks/jupyter-book/issues/455).

> Our docs will probably be the most convoluted Sphinx config you can find, but it is mainly boilerplate Sphinx config.

That is what I'm saying is terrible. I find jb's onboarding and configuration much more straightforward and [well-documented](https://jupyterbook.org/customize/config.html).

Comment (stefanv) [reply _toc.yml:3]: Yes, the markdown documents will then get executed using something like:

```
jupytext --to notebook --execute notebook.md
```

What I'm saying about Sphinx is that there really isn't configuration that needs to be done.  You just produce the executed markdown and compile the site.  I'm not against using another tool if it contributes something other than configuration for features we don't use.

Comment (stefanv) [reply _toc.yml:3]: I am happy to go with a "whomever does the work gets to decide" approach :)

Comment (jni) [reply _toc.yml:3]: Ok but it's "whoever". =P

Comment (stefanv) [reply _toc.yml:3]: Thanks, fixed.

**Comment by lagru** [comment]
I'm taking the liberty to approve and merge this in anticipation of EuroSciPy 2022. 

That does not mean I endorse this approach over the ones proposed in #59 or #66. Although I'm heavily in favor of migrating away from the ipynb format. What I like about jupytext is the nice text format that can be opened as a Python script if configured such and that the hurdle to open it as a notebook is not that big.
